units/event_log_tests.cc

```diff
@@ -0,0 +1,183 @@
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <cstdio>
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace {
+// Set of supported attribute types.
+enum AttributeType { String, Int64, VectorInt64 };
+// Specifies the importance of an Event. Events are logged based on their level
+// of importance. Eg., compile_time.csv only contains the important compile time
+// events while event.log contains all the compile time events.
+enum LogLevel { Low, Medium, High };
+
+std::string ValueToCSVString(const std::string &value) { return value; }
+
+std::string ValueToCSVString(const int64_t value) {
+  return std::to_string(value);
+}
+
+std::string ValueToCSVString(const std::vector<int64_t> &values) {
+  std::string tmp = "\"[";
+  using std::begin;
+  using std::end;
+  auto e = end(values);
+  for (auto it = begin(values); it != e; ++it) {
```

Comment:
This loop no longer needs to be generic, we should change this to a plain for loop over all indices. Something like:
```c++
for (size_t i  = 0, e = values.end(); i != e; ++i)
```